targets/curie_bsp/jerry_app/quark/main.c

```diff
@@ -0,0 +1,169 @@
+/* Copyright 2016 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <stdio.h>
+#include <stdint.h>
+#include <string.h>
+
+/* Infra */
+#include "infra/bsp.h"
+#include "infra/reboot.h"
+#include "infra/log.h"
+#include "infra/time.h"
+#include "infra/system_events.h"
+#include "infra/tcmd/handler.h"
+
+#include "cfw/cfw.h"
+/* Watchdog helper */
+#include "infra/wdt_helper.h"
+
+#include "jerry-api.h"
+#include "string.h"
+#include "jerry-port.h"
+
+#include "zephyr.h"
+#include "microkernel/task.h"
+#include "os/os.h"
+
+static T_QUEUE queue;
+
+jerry_value_t print_function;
+
+void jerry_resolve_error (jerry_value_t ret_value)
+{
+  if (jerry_value_has_error_flag (ret_value))
+  {
+    jerry_value_clear_error_flag (&ret_value);
+    jerry_value_t err_str_val = jerry_value_to_string (ret_value);
+    jerry_char_t err_str_buf[256];
+    jerry_size_t err_str_size = jerry_get_string_size (err_str_val);
+    jerry_size_t sz = jerry_string_to_char_buffer (err_str_val, err_str_buf, err_str_size);
```

Comment:
thanks for the reminder , the buffer is dynamically created regarding the string_size